### ISSUE TITLE
Strange predictions.

### ISSUE THREAD

**Issue by asmodaay** [QData/spacetimeformer#28]
Hi! Do you have any ideas about strange shift in predictions? I mean the shape of prediction is pretty well and looks like target, but the distance between context and prediction is too big.
![media_images_test_prediction_plots_34_7](https://user-images.githubusercontent.com/45363107/150631970-11a9cd35-f185-46b2-885d-f09da99ced75.png)
![media_images_test_prediction_plots_29_3](https://user-images.githubusercontent.com/45363107/150631974-af10a9fb-cb12-435b-b6f0-faa3089e7dea.png)
![media_images_test_prediction_plots_29_2](https://user-images.githubusercontent.com/45363107/150631976-77603c8b-896f-445b-a36e-aec2a0c4982e.png)

**Comment by qiyanjun** [comment]
@jakegrigsby can you check this?

**Comment by jakegrigsby** [comment]
Hmm, there's not a lot to go on here but possibly this is a normalization issue? Is that one of our datasets or are you trying your own? It doesn't seem normalized... or maybe it is normalized but is really non-stationary so that the validation data from the end of the timeseries is much larger than what was seen during training (common problem in domains where things generally go up over time... finance for example). That might lead to the correct shape but with predictions that are too low like you're seeing here. We do have an extra trick you can try to deal with this:
https://github.com/QData/spacetimeformer/blob/1ee8c8143a218b5d70a688c4929ba9bca7441914/spacetimeformer/spacetimeformer_model/spacetimeformer_model.py#L48
We have this "linear_window" concept that we don't talk about in the paper draft because we don't use it on those datasets. Basically it uses a linear combination of `linear_window` points to create an offset for the rest of the predictions. This lets the Transformer learn the shape while the linear model learns the starting point.
https://github.com/QData/spacetimeformer/blob/1ee8c8143a218b5d70a688c4929ba9bca7441914/spacetimeformer/forecaster.py#L108-L125

I put this in because LSTNet does something a lot like this and I wanted to see how much it mattered. The answer was not much, at least on the paper datasets. But it did help a lot on an exchange rates dataset that is non-stationary. 

**To summarize: make sure you are normalizing the data based on training set statistics, and then try `--linear_window 12` or 24.**

**Comment by asmodaay** [comment]
Thank a lot for your advice, i will try it and return with feedback. This is really non-stationary financial data with a high growth rate, but I had same issue with your exchange dataset but in a less degree.

**Comment by ebuss81** [comment]
Hey everyone,
first, many thanks to all your work. Its great!
I probably have a comparable question. My data is highly non-stationary and the predictions are only a straight line.  Can we assume that the high non-stationarity of the data is probably the main cause?

![prediction_plots_2002_1](https://user-images.githubusercontent.com/49711969/153025277-a4622a41-f39a-436f-8d18-15d5638b1545.png)
![e7ac0f8b-5f10-4303-a0fb-27fa73723c69](https://user-images.githubusercontent.com/49711969/153025605-ce0402a5-84cd-4c55-9dbb-5baa00a53462.png)

Best
Eddy

**Comment by meerahshivn** [comment]
![image](https://user-images.githubusercontent.com/60868672/190540465-7cb9e927-5885-44c1-8fc2-135ec0f2e096.png)
Hello, i'm a beginner on this field and this project seems very interesting to explore for my research. Well, using asos dataset, I obtained result as shown in the image without the error computation MAPE, is it something hard-coded and required to change or is it related any other issues dependency of version...please do advise me...
